### DISCUSSION TITLE
Add_self_loops doesn't change a result

### DISCUSSION THREAD

**Discussion by kenenbek** [pyg-team/pytorch_geometric#7292] [Q&A] [answered]
I have such model with self-loops

`model = GCNConv(in_channels=1, out_channels=1, add_self_loops=True, normalize=False)
`
But I don't specify any edgees here:
```
simple_edge_index = torch.tensor([[],
                           []], dtype=torch.long)

```
And I have simple input data:

`x = torch.tensor([[1], [1], [1], [1], [1]], dtype=torch.float)`

I run the model: 
`h = model(x, simple_edge_index)
`

And as a result I get: 

```
tensor([[0.],
        [0.],
        [0.],
        [0.],
        [0.]], grad_fn=<AddBackward0>)

```

Could you explain why it happened? Because I have self_loops and non-zero weights of the model 
```
OrderedDict([('bias', tensor([0.])), ('lin.weight', tensor([[0.1189]]))])
```

I expected to get something like (because of self-loops): 

```
tensor([[0.1189],
        [0.1189],
        [0.1189],
        [0.1189],
        [0.1189]], grad_fn=<AddBackward0>)

```

**Comment by rusty1s** [comment, marked as answer]
Mh, I cannot reproduce this. This works for me
```python
import torch

from torch_geometric.nn import GCNConv

conv = GCNConv(1, 1)

edge_index = torch.empty(2, 0, dtype=torch.long)
x = torch.ones(4, 1)

out = conv(x, edge_index)
print(out)
```
Can you confirm?

**Reply by rusty1s** [reply]
The issue seems to be with `normalize=False`, since it currently also disables adding self-loops :(
